libobs/obs.c

```diff
@@ -1769,6 +1787,106 @@ void obs_set_output_source(uint32_t channel, obs_source_t *source)
 	}
 }
 
+obs_source_t *obs_get_audio_track_source(uint32_t track)
+{
+	obs_source_t *source;
+	assert(track < MAX_AUDIO_MIXES);
+
+	if (track >= MAX_AUDIO_MIXES)
+		return NULL;
+
+	pthread_mutex_lock(&obs->data.audio_sources_mutex);
+	source = obs_source_get_ref(obs->data.tracks[track]);
+	pthread_mutex_unlock(&obs->data.audio_sources_mutex);
+
+	return source;
+}
+
+int obs_get_audio_track_index(obs_source_t *source)
+{
+	if (!source)
+		return -1;
+
+	int track = -1;
+
+	pthread_mutex_lock(&obs->data.audio_sources_mutex);
+
+	for (int i = 0; i < MAX_AUDIO_MIXES; i++) {
+		if (obs->data.tracks[i] == source) {
+			track = i;
+			break;
+		}
+	}
+
+	pthread_mutex_unlock(&obs->data.audio_sources_mutex);
+
+	return track;
+}
+
+void obs_set_audio_track_source(uint32_t track, obs_source_t *source)
+{
+	assert(track < MAX_AUDIO_MIXES);
+
+	if (track >= MAX_AUDIO_MIXES)
+		return;
+
+	if (!obs_is_source_audio_track(source))
+		return;
+
+	struct obs_source *prev_source;
+
+	pthread_mutex_lock(&obs->data.audio_sources_mutex);
+
+	source = obs_source_get_ref(source);
+	prev_source = obs->data.tracks[track];
+	obs->data.tracks[track] = source;
+
+	pthread_mutex_unlock(&obs->data.audio_sources_mutex);
+
+	if (source)
+		obs_source_activate(source, MAIN_VIEW);
+
+	if (prev_source) {
+		obs_source_deactivate(prev_source, MAIN_VIEW);
+		obs_source_release(prev_source);
+	}
+}
+
+void obs_check_audio_track_feedback(obs_source_t *source, bool activating)
+{
+	if (!(source->info.output_flags & OBS_SOURCE_DO_NOT_SELF_MONITOR))
+		return;
+
+	obs_data_t *s = obs_source_get_settings(source);
+	const char *s_dev_id = obs_data_get_string(s, "device_id");
+	obs_data_release(s);
+
+	const char *id = obs->audio.monitoring_device_id;
+
+	if (!s_dev_id || !*s_dev_id || !id || !*id)
+		return;
+
+	bool match = devices_match(s_dev_id, id);
+
+	if (match) {
+		if (activating) {
+			os_atomic_inc_long(&obs->data.feedback_detection);
+
+			blog(LOG_WARNING,
+			     "Prevented feedback loop between '%s' and the master audio monitor",
+			     obs_source_get_name(source));
+		} else {
+			os_atomic_dec_long(&obs->data.feedback_detection);
+		}
+	}
+}
+
+bool obs_is_source_audio_track(obs_source_t *source)
+{
+	uint32_t flags = obs_source_get_output_flags(source);
+	return (flags & OBS_SOURCE_AUDIO_TRACK) == OBS_SOURCE_AUDIO_TRACK;
```

Comment:
Following the implementation of `obs_source_is_scene`, I suggest to use `strcmp` instead of the new flag.
```suggestion
	return source && strcmp(source->info.id, audio_track_info.id) == 0;
```